packages/pyright-scip/src/treeVisitor.ts

```diff
@@ -1320,44 +1295,49 @@ export class TreeVisitor extends ParseTreeWalker {
         }
     }
 
+    private getFunctionSymbol(decl: FunctionDeclaration): ScipSymbol {
+        const declModuleName = decl.moduleName;
+        let pythonPackage = this.guessPackage(declModuleName, decl.path);
+        if (!pythonPackage) {
+            return ScipSymbol.local(this.counter.next());
+        }
+
+        const enclosingClass = ParseTreeUtils.getEnclosingClass(decl.node);
+        if (enclosingClass) {
+            const enclosingClassType = this.evaluator.getTypeOfClass(enclosingClass);
+            if (enclosingClassType) {
+                let classType = enclosingClassType.classType;
+                const pythonPackage = this.guessPackage(classType.details.moduleName, classType.details.filePath)!;
+                const symbol = Symbols.makeClass(pythonPackage, classType.details.moduleName, classType.details.name);
+                return Symbols.makeMethod(symbol, decl.node.name.value);
+            }
+            return ScipSymbol.local(this.counter.next());
+        } else {
+            return Symbols.makeMethod(Symbols.makeModule(pythonPackage, declModuleName), decl.node.name.value);
+        }
+    }
+
+    // NOTE(tech-debt): typeToSymbol's signature doesn't make sense. It returns the
```

Comment:
Symbols should be purely based on declarations.

For example: https://sourcegraph.sourcegraph.com/github.com/sourcegraph/scip-clang/-/blob/indexer/SymbolFormatter.cc?L707